components/log-viewer-webui/server/src/routes/query.ts

```diff
@@ -1,58 +1,66 @@
+import {FastifyPluginAsync} from "fastify";
 import {StatusCodes} from "http-status-codes";
 
+import {TypeBoxTypeProvider} from "@fastify/type-provider-typebox";
+import {Type} from "@sinclair/typebox";
+
 import settings from "../../settings.json" with {type: "json"};
-import {EXTRACT_JOB_TYPES} from "../DbManager.js";
+import {
+    EXTRACT_JOB_TYPES,
+    QUERY_JOB_TYPE,
+} from "../DbManager.js";
 
 
 /**
  * Creates query routes.
  *
- * @param {import("fastify").FastifyInstance | {dbManager: DbManager}} fastify
- * @param {import("fastify").FastifyPluginOptions} options
- * @return {Promise<void>}
+ * @param app
  */
-const routes = async (fastify, options) => {
-    fastify.post("/query/extract-stream", async (req, resp) => {
+const routes: FastifyPluginAsync = async (app) => {
+    const fastify = app.withTypeProvider<TypeBoxTypeProvider>();
+
+    fastify.post("/query/extract-stream", {
+        schema: {
+            body: Type.Object({
+                extractJobType: Type.Enum(QUERY_JOB_TYPE),
+                logEventIdx: Type.Integer(),
+                streamId: Type.String({minLength: 1}),
+            }),
+        },
+    }, async (req, resp) => {
         const {extractJobType, logEventIdx, streamId} = req.body;
         if (false === EXTRACT_JOB_TYPES.includes(extractJobType)) {
             resp.code(StatusCodes.BAD_REQUEST);
             throw new Error(`Invalid extractJobType="${extractJobType}".`);
         }
 
-        if ("string" !== typeof streamId || 0 === streamId.trim().length) {
```

Comment:
The empty string check is now done by `Type.String({minLength: 1})` though not trimmed. If needed, we can define a Regex check instead.